acvm-repo/acvm/tests/solver.rs

```diff
@@ -910,4 +1342,75 @@ proptest! {
         let (lhs, rhs) = prop_assert_zero_l(and_op, zero, x);
         prop_assert_eq!(lhs, rhs);
     }
+
+    #[test]
+    fn poseidon2_permutation_matches_external_impl(inputs in proptest::collection::vec(field_element(), 4)) {
+        let (result, expected_result) = run_both_poseidon2_permutations(inputs);
+        prop_assert_eq!(result, expected_result)
+    }
+
+    #[test]
+    fn sha256_injective(inputs_distinct_inputs in any_distinct_inputs(None, 0, 32)) {
+        let (inputs, distinct_inputs) = inputs_distinct_inputs;
+        let (result, message) = prop_assert_injective(inputs, distinct_inputs, 32, None, sha256_op);
+        prop_assert!(result, "{}", message);
+    }
+
+    #[test]
+    fn sha256_compression_injective(inputs_distinct_inputs in any_distinct_inputs(None, 24, 24)) {
+        let (inputs, distinct_inputs) = inputs_distinct_inputs;
+        if inputs.len() == 24 && distinct_inputs.len() == 24 {
+            let (result, message) = prop_assert_injective(inputs, distinct_inputs, 8, None, sha256_compression_op);
+            prop_assert!(result, "{}", message);
+        }
+    }
+
+    #[test]
+    fn blake2s_injective(inputs_distinct_inputs in any_distinct_inputs(None, 0, 32)) {
+        let (inputs, distinct_inputs) = inputs_distinct_inputs;
+        let (result, message) = prop_assert_injective(inputs, distinct_inputs, 32, None, blake2s_op);
+        prop_assert!(result, "{}", message);
+    }
+
+    #[test]
+    fn blake3_injective(inputs_distinct_inputs in any_distinct_inputs(None, 0, 32)) {
+        let (inputs, distinct_inputs) = inputs_distinct_inputs;
+        let (result, message) = prop_assert_injective(inputs, distinct_inputs, 32, None, blake3_op);
+        prop_assert!(result, "{}", message);
+    }
+
+    #[test]
+    fn keccak256_injective(inputs_distinct_inputs in any_distinct_inputs(Some(8), 0, 32)) {
+        let (inputs, distinct_inputs) = inputs_distinct_inputs;
+        let (result, message) = prop_assert_injective(inputs, distinct_inputs, 32, Some(32), keccak256_op);
+        prop_assert!(result, "{}", message);
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5689): doesn't fail with an error, returns constant output
+    #[test]
+    #[should_panic(expected = "Test failed: not injective")]
```

Comment:
Nit: this message seems off? If this is panicking as expected does that mean #5689 is not an issue?

Reply:
I added some detail: the test for injectivity fails, but we want the invocation of the solver to be the part that fails (with a user-level error).